skfda/ml/_neighbors_base.py

```diff
@@ -537,6 +545,31 @@ def predict_proba(
             self._estimator.predict_proba(X_dist)
         )
 
+    def fit(
+        self: SelfTypeClassifier,
+        X: Input,
+        y: TargetClassification,
+    ) -> SelfTypeClassifier:
+        """
+        Fit the model using X as training data and y as responses.
+
+        Args:
+            X: Training data. FDataGrid
+                with the training data or array matrix with shape
+                [n_samples, n_samples] if metric='precomputed'.
+            y: Training data. FData with the training respones (functional
+                response case) or array matrix with length `n_samples` in
+                the multivariate response case.
+
+        Returns:
+            Self.
+
+        Note:
+            This method adds the attribute `classes_` to the classifier.
+        """
+        self.classes_ = _classifier_get_classes(y)[0]
+        return super().fit(X, y)
```

Comment:
Shouldn't the passed `y` be the parsed one?

Reply:
In the KNeighbours classes there really is no encoding and the predicted values (after `predict`), have the correct labels in `classes_`.
However, this is not the case for logistic regression, because the underlying `fit` method receives `y_ind`, so the predicted values are the encoded index of the labels. 

See
https://github.com/GAA-UAM/scikit-fda/blob/566e3e3f8e87e4ccf51cf71c538da57feefe4d74/skfda/ml/classification/_logistic_regression.py#L108-L182
where the resulting `self._mlvr` is the wrapped predictor, but it gets fitted with the indices, as shown in L174, instead of the actual classes.

I propose to solve this problem of logistic regression in another pull request. I will modify the test script to exclude logistic regression for now, and include the assertion between `classes_` and the unique values of `predict(X)`.